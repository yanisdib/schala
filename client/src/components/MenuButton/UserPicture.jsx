import PropTypes from 'prop-types';
import styled from 'styled-components';

import defaultPicture from '../../assets/images/blank_profile.jpg';


const UserPicture = styled.div`
    width: 30px;
    height: 30px;
    background-image: url(${props => props.picture});
    background-size: cover;
    background-position: center;
    background-color: white;
    border-radius: 1.5rem;
    margin-left: 10px;
`;


UserPicture.defaultProps = { picture: defaultPicture }

UserPicture.propTypes = { picture: PropTypes.string }


export default UserPicture;