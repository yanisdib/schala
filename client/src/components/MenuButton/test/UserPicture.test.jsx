import { shallow } from 'enzyme';

import UserPicture from '../../../components/MenuButton/UserPicture';


describe('UserPicture component', () => {
    let wrapper;
    const { picture } = UserPicture.defaultProps;

    beforeEach(() => wrapper = shallow(<UserPicture />))

    it('renders properly', () => expect(wrapper).toMatchSnapshot())

    it('renders with default picture props', () => expect(picture).toEqual('blank_profile.jpg'))
})