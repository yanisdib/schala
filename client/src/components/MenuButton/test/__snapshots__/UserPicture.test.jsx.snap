// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPicture component renders properly 1`] = `
<div
  className="sc-bdfBQB drxxLc"
/>
`;
